lib/faktory_worker/job.ex

```diff
@@ -152,12 +152,12 @@ defmodule FaktoryWorker.Job do
         opts
       )
     else
-      message = %Broadway.Message{
-        acknowledger: {FaktoryWorker.PushPipeline.Acknowledger, :push_message, []},
-        data: {pipeline_name, payload}
-      }
-
-      Broadway.push_messages(pipeline_name, [message])
+      # TODO:: do we want to split pipelines into multiple threads???
+      # It was implemented with a Broadway & Producers <--> processes model.
+      # But Faktory is a job server and do we want to care and build one more not needed layer???
+      opts
+      |> faktory_name()
+      |> push(payload)
```

Comment:
Switching from `Broadway.push_messages` to `push` is going to change the return of this function.  From the PR discussion, I gather we are planning on doing a version change.  
1) Are there additional changes needed to this code base to signal that change?  
2) Does module/package documentation need to be updated?

Reply:
Not for this PR, just documenting a thought - but if we can return a Job with the faktory job id here, that would be a great gateway into being able to check on the status of a job.

Reply:
`Broadway.push_messages` and `Sandbox.enqueue_job` both always return `:ok`, so I think it should be OK to have this return `{:ok, job_id}` instead.

+1 for updating documentation as well

Reply:
it looks like we don't need perform_async/3 at all and we can return same result for any call. private function `handle_push_result` prepares result and it returns {:ok, job}.
I am not sure is it better to change to {:ok, job_id} or leave it as is. `job` contains a bit more info, see sample below:
```
{:ok,
 %{
   args: [%{hey: "there!"}],
   jid: "cc7cb6ec723b1e1f35815ba1",
   jobtype: "FaktoryWorker.DefaultWorker"
 }}

```

Reply:
It should be OK to return it as-is. Can you also update `Sandbox.enqueue_job` to return a similar value?

Reply:
@superhawk610 we have changed return to {:ok, job}, but I checked our app and I see it always expect :ok.
Question is -> revert changes and return :ok as before, change app and it's tests to have support of {:ok, job). or 3rd option: add configuration option and return optionally :ok to keep back compatibilities and return by default {:ok, job}.

Reply:
Let's change our app to accept `{:ok, job}`

Reply:
Make sure that this is mentioned in the upgrade guide as well, as it's a breaking change.